etna/records/templatetags/records_tags.py

```diff
@@ -10,12 +11,24 @@
 
 
 @register.simple_tag
-def record_url(record: Union[Record, Dict[str, Any]]) -> str:
+def record_url(
+    record: Union[Record, Dict[str, Any]], is_editorial: bool = False
+) -> str:
     """
     Return the URL for the provided `record` dict; which could either be a
     full/transformed result from the fetch() endpoint, OR a raw result from
     the search() endpoint.
     """
+    if is_editorial and settings.FEATURE_RECORD_LINKS_GO_TO_DISCOVERY:
+        if iaid := record.get("iaid"):
+            return f"https://discovery.nationalarchives.gov.uk/details/r/{iaid}"
+        if url := record.get("sourceUrl"):
+            return url
```

Comment:
Does this work for record type?

Reply:
Same as previous comment above for record type.

Reply:
@TNA-Allan What I mean is: look at the existing code just below... it's doing the exact same thing: https://github.com/nationalarchives/ds-wagtail/blob/79d0019fcd0fbed23d82f0fbfca88d9961e6f84f/etna/records/templatetags/records_tags.py#L44-L45

If we are changing the logic of where the sourceURL is pulled from, then that is a separate PR. Does this make sense?

Reply:
@TNA-Allan I've refactored to avoid repeating lines unnecessarily. Please mark as resolved if you are happy :)

Reply:
@ababic I think it tricky to understand and debug this method since it caters to 2 different kinds in instances for Record.
After refactoring, it looks like more conditions are introduced :)

Reply:
@ababic I think it tricky to understand and debug this method since it caters to 2 different kinds in instances for Record.
After refactoring, it looks like more conditions are introduced :)